CONTRIBUTING.md

```diff
@@ -0,0 +1,96 @@
+# Contributing
+
+By contributing, you are agreeing that we may redistribute your work under this [license](LICENSE).
+
+# Table of contents
+- [Getting started as a developer](#getting-started-as-a-developer)
+- [The development life cycle](#the-development-life-cycle)
+   * [Jira](#jira)
+   * [Developing and testing locally](#developing-and-testing-locally)
+   * [Testing remotely](#testing-remotely)
+   * [Code review](#code-review)
+- [Code style](#code-style)
+
+# Getting started as a developer
+
+If you are onboarding to the RECOVER project as a developer, we provide [onboarding documentation via Confluence](https://sagebionetworks.jira.com/wiki/spaces/IBC/pages/2808053761/RECOVER+ETL#Getting-Started). There you will find everything you need to know about gaining write access to this GitHub repository and our AWS environments (and more)! An overview of the system architecture is [here](https://sagebionetworks.jira.com/wiki/spaces/IBC/pages/2741075969/RMHDR-48+RECOVER+ETL+Design).
+
+# The development life cycle
+
+To keep work organized and predictable, developers follow specific development patterns. For a high level overview of development patterns within the DPE team, see the [DPE Handbook](https://sagebionetworks.jira.com/wiki/spaces/IBC/pages/2667905251/DPE+handbook).
+
+## Jira
+
+Before there was work, there was the _work request_. Work requests a.k.a. tickets for this repository are tracked in the [ETL Jira project](https://sagebionetworks.jira.com/jira/software/c/projects/ETL/boards/190). No development work should be done without first having a Jira ticket which outlines the context and goals of the work to be done.
+
+## Developing and testing locally
+
+1. Clone this repository (do not fork). Create a feature branch off the `main` branch. Branch names must contain the identifier of the corresponding Jira ticket (e.g., `etl-574`). Optionally, add keywords describing the work that is being done to the branch name (e.g., `etl-574-add-contributing-doc`).
+
+> [!IMPORTANT]
+> Keep branch names as short as possible. We use the branch name to namespace resources in AWS (as described below). Since AWS imposes limits on the name length of some resources, the deployment may fail if the branch name is too long.
+
+2. We use [pipenv](https://pipenv.pypa.io/en/latest/) to manage our development environment. Simply run
+
+```
+pipenv shell
+pipenv install --dev
+```
+
+from the repository root to create a virtual environment and install all dependencies.
```

Comment:
The formatting ended up a little poor here when looking at the rendered md - Ideally this should maintain the same indent level:
![image](https://github.com/Sage-Bionetworks/recover/assets/17128019/8e46980d-ff64-4e3a-ad96-830b8c3dfb76)

Reply:
I don't know if it's possible to put code blocks at an indent level, so I moved the code block to the bottom of point 2 so the formatting is less jarring.